utils/grpc/graceful_termination.go

```diff
@@ -29,7 +29,7 @@ func GracefulTerminateDetect() {
 // GracefulTerminateExec performs a default list of tasks to clean up
 // Override this function in your package to do more.
 func GracefulTerminateExec(grpcServer *grpc.Server) {
-	stopped := make(chan struct{})
+	stopped := make(chan struct{}, 1)
```

Comment:
I don't quite get this. Are you afraid that `grpcServer.GracefulStop()` is stuck?

`<-stopped` will be unblocked as soon as it's closed.

Reply:
@taiyangc I agree with that. However there is an edge case scenario.

Scenario: We have initiated graceful termination. Lets say timeout happens first. -> this calls force stop()
But at the same time graceful termination is running but has not completed.

Then at some point graceful termination completes -> nothing is there to receive so it is always blocking and goroutine does not die.
Now its up to the stop() function if it kills the whole thing; then we are good. Otherwise, we got ourselves goroutine leak.

Reply:
"nothing is there to receive so it is always blocking and goroutine does not die"

The `close(stopped)` will always finish regardless if there is a receiving end.

Reply:
An example: https://play.golang.org/p/hBp6ASYvhxC

Reply:
@taiyangc 
Perhaps I am on a different page. I was thinking more of this example: https://play.golang.org/p/fV2qwWWeiS6
`close(stopped)` was not called here when graceful termination goroutine did not complete. It did not matter though because the main thread exited out. I guess we can close this out.